FLT/DivisionAlgebra/Finiteness.lean

```diff
@@ -172,8 +173,75 @@ lemma antidiag_mem_C {β : D_𝔸ˣ} (hβ : β ∈ ringHaarChar_ker D_𝔸) :
 
 end Aux
 
-lemma compact_quotient : CompactSpace (ringHaarChar_ker D_𝔸 ⧸
-  (MonoidHom.range (incl K D)).comap (ringHaarChar_ker D_𝔸).subtype) := sorry
+/-- The inclusion of `ringHaarChar_ker D_𝔸` into the product space `D_𝔸 × D_𝔸ᵐᵒᵖ`. -/
+def incl₂ : ringHaarChar_ker D_𝔸 → Prod D_𝔸 D_𝔸ᵐᵒᵖ :=
+  fun u => Units.embedProduct D_𝔸 (Subgroup.subtype (ringHaarChar_ker D_𝔸) u)
+
+/-- An auxillary set used in the proof of compact_quotient'. -/
+def M : Set (ringHaarChar_ker D_𝔸) := Set.preimage (incl₂ K D)
```

Comment:
`M` should be in the `Aux` namespace. 

In fact I now suspect that this `Aux` namespace is a bad idea and that we should just make all these auxiliary definitions private, but maybe I'll just deal with all this myself.